SolHashing2P1.py

```diff
@@ -0,0 +1,28 @@
+"""
+Given an array of integers nums and an integer k, return the total number of subarrays whose sum equals to k.
+A subarray is a contiguous non-empty sequence of elements within an array.
+Time Complexity : O(N) where N = length of nums
+Space Complexity : O(N) 
+Did this code successfully run on Leetcode : Yes
+Any problem you faced while coding this :
+Yes. I felt like this is like if know you it, you know it!! 
```

Comment:
That's why these patterns are covered in class. It will be easier in future for you to think in this direction if necessary